modular_skyrat/modules/customization/modules/clothing/clothing.dm

```diff
@@ -19,6 +19,129 @@
 	var/greyscale_config_worn_taur_paw
 	var/greyscale_config_worn_taur_hoof
 
+	var/static/has_taur_sprite = list()
+	var/static/has_taur_snake_sprite = list()
+	var/static/has_taur_paw_sprite = list()
+	var/static/has_taur_horse_sprite = list()
+
+	var/static/has_taur_sprite_suit = list()
+	var/static/has_taur_snake_sprite_suit = list()
+	var/static/has_taur_paw_sprite_suit = list()
+	var/static/has_taur_horse_sprite_suit = list()
+
+/obj/item/clothing/Initialize(mapload)
+	. = ..()
+	handle_taur_sprites()
+
+// Convert this to obj/item if you need to make a item sprite. It'll become more intensive due to it, but yeah
+
+/**
+ * Called in /obj/item/clothing/Initialize().
+ *
+ * Quits instantly if mutant_variants has STYLE_TAUR_ALL, then checks to see if the clothing's icon state has an alt sprite for taurs. If no, it then checks to see if
+ * the icon state EXPLICITELY has no taur sprites. If no, it then checks 3 DMI files for the icon_state, and if it appears in any of them, has_taur_sprite is set to
+ * true for that icon state, then it sets the more specific list to true for that state, and finally the specific mutant_variants bitflag for the clothing if the
+ * clothing doesnt already have it.
+ *
+ * Finally, if all of the conditions are false, has_taur_sprite for that icon state will be set to false.
+ *
+ * DO NOT RELY ON THIS PROC TO DO YOUR WORK FOR YOU. If you are adding a new item, or a sprite, you SHOULD be manually setting your mutant_variants flags, otherwise the for
+ * loop will be ran and cause slight performance loss.
+ */
+/obj/item/clothing/proc/handle_taur_sprites() //niko-if i do not come back in a week and entirely refactor how this works gitban me
+	if (istype(src, /obj/item/clothing/suit))
+		handle_taur_sprites_for_suits()
+		return
+	if (!(istype(src, /obj/item/clothing/under))) //remove this if we ever make a taur sprite for anything else, this is for performance
+		return
+	if (mutant_variants & STYLE_TAUR_ALL)
+		return
+	if (has_taur_sprite[icon_state] == HAS_TAUR_SPRITE)
+		if ((has_taur_snake_sprite[icon_state] == HAS_TAUR_SPRITE) && (!(mutant_variants & STYLE_TAUR_SNAKE)))
+			mutant_variants |= STYLE_TAUR_SNAKE
+		if ((has_taur_horse_sprite[icon_state] == HAS_TAUR_SPRITE) && (!(mutant_variants & STYLE_TAUR_HOOF)))
+			mutant_variants |= STYLE_TAUR_HOOF
+		if ((has_taur_paw_sprite[icon_state] == HAS_TAUR_SPRITE) && (!(mutant_variants & STYLE_TAUR_PAW)))
+			mutant_variants |= STYLE_TAUR_PAW
+		return // If we already know this icon state has a taur sprite, skip the for loop and take from the cache
+	else if (has_taur_sprite[icon_state] == HAS_NO_TAUR_SPRITE)
```

Comment:
leaving these as `(has_taur_sprite[icon_state] == HAS_NO_TAUR_SPRITE)` instead of converting to `(!(has_taur_sprite[icon_state] == HAS_NO_TAUR_SPRITE)` is preferred since the latter will pass if `icon_state` is null.

Reply:
This could probably just be FALSE instead of `HAS_NO_TAUR_SPRITE`, this doesn't really need to be a define, it's clear enough.